spec/supervisor_spec.rb

```diff
@@ -161,4 +161,117 @@ def connect task, core_versions:, sxl_version:
       end
     end
   end
+
+  describe '#proxy creation' do
+    context 'when site has TLC SXL in configuration' do
+      it 'creates TLCProxy for configured TLC sites' do
+        # Test the site settings determination logic directly
+        supervisor = RSMP::Supervisor.new(
+          supervisor_settings: {
+            'port' => 13113,
+            'sites' => {
+              'TLC001' => { 'sxl' => 'tlc', 'type' => 'tlc' }
+            },
+            'guest' => { 'sxl' => 'tlc', 'type' => 'tlc' }
+          },
+          log_settings: log_settings
+        )
+        
+        # The site_id_to_site_setting method should return the tlc config
+        site_settings = supervisor.site_id_to_site_setting('TLC001')
+        expect(site_settings['sxl']).to eq('tlc')
+        
+        settings = {
+          supervisor: supervisor,
+          ip: '127.0.0.1',
+          port: 12345,
+          socket: double('socket'),
+          stream: double('stream'),
+          protocol: double('protocol'),
+          logger: double('logger'),
+          archive: double('archive')
+        }
+        
+        # Test the logic that would be used in accept_connection
+        if site_settings && site_settings['type'] == 'tlc'
```

Comment:
a conditional does not make sense here, as settings is being set up as part of the test?